app/controllers/your-psp/post-toggle-worldpay-3ds-flex.controller.js

```diff
@@ -11,6 +11,10 @@ module.exports = async function toggleWorldpay3dsFlex (req, res, next) {
   const accountId = req.account.gateway_account_id
   const toggleWorldpay3dsFlex = req.body['toggle-worldpay-3ds-flex']
 
+  if (req.account.type === 'live'){
+    next(new Error(`Cannot toggle Worldpay 3DS flex for live gateway account: ${req.account.external_id}`))
```

Comment:
`next` is middleware, so it's our preferred way to handle unexpected errors.  
`next` will sort of intercept and go straight to the error handler.
This is an error that should not happen, adding it as protection.

Before, we used to have errors handled in different places.  We now try to have everything handled in the error handler.

Please let me know if you have any more questions about.

Reply:
Drive by from someone that's spend too much of their time looking at node.

Just to expand on Iqbal's answer, `next` in this context (and all middleware) is just a normal function which has been passed into our controller as an argument. In this case it's the Express framework that will be invoking our controller and that's who will pass the `next` function in.

A simple example might be

```javascript
function nextFunctionUsuallyDefinedByExpress() {
  console.log('The next method is managed and provided by Express')
}

function myController(next) {
  next()

  // calling the next callback will add it to the top of the stack, but it doesn't do anything magical, the code after it will still run
  console.log('The code in the controller will still run')
}

myController(nextFunctionUsuallyDefinedByExpress)
```

Output
```
> The next method is managed and provided by Express
> The code in the controller will still run
```

For this reason, you'll often see the function exited (via `return`) after passing an error through to next as you don't want to do anything after you've told Express to continue the middleware stack (which I can see has now been done in the update!).

```javascript
function nextFunctionUsuallyDefinedByExpress() {
  console.log('The next method is managed and provided by Express')
}

function myController(next) {
  next()
  return
  console.log('The code in the controller will still run')
}

myController(nextFunctionUsuallyDefinedByExpress)
```

Output
```
> The next method is managed and provided by Express
```

Reply:
thanks @sfount  - v helpful explanation